src/prefect/testing/fixtures.py

```diff
@@ -346,6 +348,23 @@ def asserting_events_worker(monkeypatch) -> Generator[EventsWorker, None, None]:
         worker.drain()
 
 
+@pytest.fixture
+async def events_pipeline(asserting_events_worker: EventsWorker):
+    class AssertingEventsPipeline(EventsPipeline):
+        def sync_process_events(self):
+            asyncio.run(self.process_events())
```

Comment:
this seems like it's working fine but accepting any/all/better sync interfaces for this

Reply:
I think you could use the `sync_compatible` decorator if you wanted and then call it with `_sync=True` as needed within tests